### ISSUE TITLE
[feature request] Allow filter by `agent.group` in `Wazuh Modules/Security events`

### ISSUE THREAD

**Issue by carlosdams** [wazuh/wazuh#14225]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.x|Security events| Manager  |N/A|Any|

**Description:**
Allow filtering by group of agents in Security events (screenshots below)

Wazuh 4.2.x :
![image](https://user-images.githubusercontent.com/35348824/178576712-e1f389c8-006b-47fa-847e-f808105d2ca2.png)

Wazuh 4.3.x:
![image](https://user-images.githubusercontent.com/35348824/178576954-a3421910-17c3-4b97-8b94-b5641e2d21ad.png)

**Comment by nzalev** [comment]
Just ran into this today. This would be very nice.

**Comment by joseraeiro** [comment]
It would be nice indeed, to have the possibility of filtering per Agent Group.

**Comment by MACGadams** [comment]
Yo, this would be really nice <3

**Comment by sapient-security** [comment]
Also would like to see this functionality as it's critical for MSSPs and MSPs. Which PM do we need to take to lunch to get this added in the next sprint 😂 Thanks for all the hard work Team Wazuh!!

**Comment by joseraeiro** [comment]
There is a workaround for this issue. One can create a label for the group for which we need to filter in Groups like this:

<img width="319" alt="2023-04-19 11_13_51-Wazuh - Wazuh" src="https://user-images.githubusercontent.com/8979838/233044557-c075969e-a1d7-47a1-9a94-7c4688451d7a.png">

And then filter it out like so:

<img width="307" alt="2023-04-19 11_11_49-Wazuh - Wazuh" src="https://user-images.githubusercontent.com/8979838/233044666-8bab15cd-e847-462c-bf04-fcdaac79c42f.png">

The only inconvenience with this approach is that one must have only one label created per agent (if you define several labels (groups) for one agent Wazuh will pick one randomly)

**Comment by JackThird** [comment]
+1

**Comment by sis6326r** [comment]
+1

**Comment by SecurityArsenal** [comment]
+1  I am surprised this has not been resolved yet.  This makes it so I can't use this product to host all events from different clients and be able to report on a client (grouped by name), without leaking other client information.  This product has so much potential!

**Comment by daryltamo** [comment]
+1

**Comment by daryltamo** [comment]
Can we have a filter on agent status or agent registration date too?

**Comment by marcinliszka12** [comment]
This feature is really needed if you have a lot of agents.

**Comment by vh54** [comment]
+1

**Comment by hformaggi** [comment]
Hi guys,

I encountered the same issue and found a solution that worked to me.

I created a custom tag called **test-flag** and noticed a message alerting me about the index refresh appeared in the field name.

![wazuh-index-message](https://github.com/user-attachments/assets/2704eefd-208d-410a-b1a6-594961209274)

To resolve the issue, I followed these steps:

1. On the left hamburger menu, click **Management > Stack Management**.  
   ![wazuh-stack-management](https://github.com/user-attachments/assets/02ad3909-ea65-4727-ba05-ba2ffc324a48)

2. Click on **Index Patterns**.  
   ![wazuh-index-patterns](https://github.com/user-attachments/assets/a489a016-e78d-4649-853e-967acc218818)

3. From the list, select the index pattern **wazuh-alerts-**.  
   ![wazuh-index-patterns-list](https://github.com/user-attachments/assets/9c0761b6-a839-40bc-87eb-011cc8df3010)

4. Click the refresh icon to update the field list. *Note: This action resets the popularity counter of each field.*  
   ![wazuh-refresh-index-patterns](https://github.com/user-attachments/assets/f7dfe967-905e-4509-b83f-3272e9fd950b)

5. Refresh the discovery page and check if the warning has disappeared.  
   ![wazuh-without-alert](https://github.com/user-attachments/assets/6475c549-96d5-4668-bab4-a7d5b81f4b56)

6. You should now be able to use your custom tag in the filter field.  
   ![wazuh-filter-test-flag](https://github.com/user-attachments/assets/f6bb862a-5af9-4068-a07f-1a061483c45c)

That's it! I hope this helps :)

**Comment by robinarz** [comment]
Hi, any updates on that feature ?
Thanks by advance.

**Comment by MrCr0a** [comment]
Yep, same here, I don't see the point of group if you can't use them. Thanks

**Comment by kayo77** [comment]
+1

**Comment by pcreadycl** [comment]
Hi, any update from this?
